client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -106,8 +108,32 @@ def _handle_machine_list(self, attr_values, job_info):
         if machine_list:
             if attr_values["machine_list_deny"]:
                 job_info.Blacklist = machine_list
+                job_info.Whitelist = None
```

Comment:
Shouldn't this be resolved by the job object automatically?

- If both are empty, both are `None`.
- If one is filled other is `None`.
- If both are filled crash.

Reply:
```
    def serialize(self):
        """Return all data serialized as dictionary.

        Returns:
            OrderedDict: all serialized data.

        """
        output = {}
        for field_item in fields(self):
            self._fill_serialize_value(
                field_item.name, getattr(self, field_item.name), output
            )

        if output["Whitelist"] and output["Blacklist"]:
            raise ValueError(
                "Whitelist and Blacklist can't be set at the same time."
            )

        if output["Whitelist"]:
            output["Blacklist"] = None
        else:
            output["Whitelist"] = None
        return output
```

Reply:
What about this comment?